pkgs/tools/filesystems/dwarfs/default.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv
+, bison
+, boost
+, cmake
+, double-conversion
+, fetchFromGitHub
+, fmt_8
+, fuse3
+, gflags
+, git
+, glog
+, gtest
+, jemalloc
+, lib
+, libarchive
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkg-config
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # Needs tagged git repo for generating some files
+  # leaveDotGit alone isn't enough and breaks reproducible hashing
+  preConfigure = ''
+    git init
+    git add .
+    git -c user.name=root -c user.email=root@localhost commit -m "v${version}" -q
+    git -c user.name=root -c user.email=root@localhost tag "v${version}" -m "v${version}"
+  '';
+
+  # ...but the files are still kinda ugly, so let's make the
+  # data prettier for wherever it gets used ;)
+  postConfigure = ''
+    sed -i \
+      -e 's/\(PRJ_GIT_REV = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_DESC = \).*/\1"${version}";/' \
+      -e 's/\(PRJ_GIT_BRANCH = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_ID = \).*/\1"v${version}";/' \
+      ../src/dwarfs/version.cpp
+  '';
+
+  cmakeFlags = [
+    "-DPREFER_SYSTEM_ZSTD=ON"
+    "-DPREFER_SYSTEM_XXHASH=ON"
+    "-DPREFER_SYSTEM_GTEST=ON"
+
+    "-DWITH_LEGACY_FUSE=ON"
```

Comment:
So by my understanding of the project's CMakeLists file, FUSE 3 can build the driver for FUSE 2 also. Maybe we should put this behind a feature flag so people who don't want the FUSE 2 driver in addition to the FUSE 3 can disable it? (Should it be enabled by default? Is there a common use case for FUSE 2 that makes it so it should?)

Reply:
I based that on @tpwrules's derivation (found [here](https://github.com/tpwrules/nixpkgs/blob/1fbed16754ff6a1633cfab69e80ffcdcff7274cd/pkgs/tools/filesystems/dwarfs/default.nix#L84)) which talks about how (most) other FUSE filesystems include a FUSE legacy driver by default too. I haven't went out to confirm it myself, but I trust them.

Reply:
Why do we need fuse2 at all? If it is legacy we might as well not include it until someone needs it.

Reply:
I've commented out the FUSE2 driver for now.

Reply:
@tpwrules What's your opinion on the legacy FUSE driver option; was there a reason you originally included it in your derivation?

Reply:
A somewhat random survey of existing FUSE filesystems in nixpkgs and because it was supported by dwarfs. If you don't think it's necessary to have by default I have no problem having it removed.

Reply:
If we do it in other packages, maybe it's worth enabling by default, but make it configurable via an override flag?

Reply:
IMO that seems too complicated for now. It should either have both 2 and 3, or just 3. We can see if a legacy usecase presents itself down the line.

Reply:
Not sure, then. If other packages have it unconditionally, it may be good to follow their lead? /shrug